docs/docs/workflows/upgrade.md

```diff
@@ -1,44 +1,50 @@
 # Upgrade your cluster
 
-Constellation provides an easy way to upgrade to the next release.
-This involves updating the CLI, choosing a new VM image to use for all nodes in the cluster, and updating the cluster's expected measurements.
+Constellation provides an easy way to upgrade all components of your cluster, without disrupting it's availability.
+Specifically you can upgrade the Kubernetes version, the nodes' image and the Constellation microservices.
+You configure the desired versions in your local Constellation configuration and trigger upgrades with the `upgrade apply` command.
+To learn about available versions you use the `upgrade check` command.
+Which versions are available depends on the CLI version you are using.
 
 ## Update the CLI
 
-New features and bug fixes are added to the CLI with every release. To use them, update the CLI to the latest version by following the instructions in the [installation guide](../getting-started/install.md).
+Each CLI comes with a set of microservice and Kubernetes versions
+It's recommend to always use the most up-to-date CLI.
+However, there are two constraints to this recommendation:
+1. Don't upgrade more than one minor version at a time.
+For example, if you are currently on CLI version v2.6, you should upgrade to v2.7, upgrade your cluster and only then continue upgrading the CLI after.
+2. Don't upgrade to a CLI version that doesn't support your current Kubernetes version.
+If your current Kubernetes version is dropped from the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
```

Comment:
I find this hard to understand. I tried rephrasing it to what you should do instead of what you shouldn't do:
```suggestion
Each CLI comes with a set of supported microservice and Kubernetes versions.
Most importantly, a given CLI version can only upgrade a cluster of the previous minor version, but not older ones.
This means that you have to upgrade your CLI and cluster one minor version at a time.

For example, if you are currently on CLI version v2.6 and the latest one is v2.8, you should

* upgrade the CLI to v2.7,
* upgrade the cluster to v2.7,
* and only then continue upgrading the CLI (and the cluster) to v2.8 after.

Also note that if your current Kubernetes version isn't supported anymore by the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
```
What do you think?

Reply:
I like it a lot. Changed it.